airbyte-webapp/src/packages/cloud/views/workspaces/WorkspaceSettingsView/WorkspaceSettingsView.tsx

```diff
@@ -77,22 +77,35 @@ export const WorkspaceSettingsView: React.FC = () => {
                     />
                   )}
                 </Field>
-                <Buttons>
+                <Label className={styles.formItem}>
+                  <FormattedMessage id="settings.generalSettings.form.advancedMode.label" />
+                </Label>
+                <Field name="advancedMode">
+                  {({ field }: FieldProps<boolean>) => (
+                    <LabeledSwitch
+                      label={<AdvancedModeSwitchLabel />}
+                      checked={field.value}
+                      onChange={() => setFieldValue(field.name, !field.value)}
+                    />
+                  )}
+                </Field>
+
+                <div className={classNames(styles.formItem, styles.buttonGroup)}>
                   <Button type="button" secondary disabled={!dirty} onClick={() => resetForm()}>
```

Comment:
While testing locally, I noticed that if I toggle Advanced mode to be enabled, and I click `save changes` and wait for that to complete, and then I click `cancel`, it sets `Advanced mode` back to disabled.

As a user, I was expecting `cancel` to just get rid of any changes I had made since the last time I clicked `save changes`, so this was unexpected for me. Is this the intended behavior?

Reply:
After playing with this a bit more, I've realized that if I enabled advanced mode, then navigate to another page, and come back to this settings page, then I see the opposite behavior -- I can disable advanced mode, click save, and then if I click `cancel` it sets Advanced mode back to enabled.

So I think the part that's causing me confusion is that I expect `save changes` to set the values I've selected as the "non-dirty" values for this form, and I expect the `cancel` button to be disabled when I do so (but it is not). But what is actually happening is it seems the "non-dirty" values for the form are whatever was set when I initially loaded the page.

This likely isn't an issue that was introduced by any of your changes in your PR, but rather we may just be using Formik incorrectly

Reply:
> As a user, I was expecting cancel to just get rid of any changes I had made since the last time I clicked save changes, so this was unexpected for me.

Whoa, I didn't expect that, either. A little further testing shows that this form's save/cancel buttons do the exact same thing with changes to the workspace name, so it's definitely a form-wide issue.

Reply:
@ambirdsall want to create an issue for that?

Reply:
Done: https://github.com/airbytehq/airbyte/issues/16149